### ISSUE TITLE
[valid-types] `extends` property causes syntax error in namepath

### ISSUE THREAD

**Issue by reduckted** [gajus/eslint-plugin-jsdoc#1455]
## Expected behavior

A property called `extends` in an `@typedef` should be valid. For example, defining a type that represents a TypeScript configuration with an [`extends`](https://www.typescriptlang.org/tsconfig/#extends) property.

## Actual behavior

A property called `extends` in an `@typedef` causes a syntax error:
```
Syntax error in namepath: extends  jsdoc/valid-types
```

This was valid in `54.0.0` but fails in `54.1.0`.

## ESLint Config

```js
import jsdoc from 'eslint-plugin-jsdoc';


/** @type {import('eslint').Linter.Config} */
export default {
    files: ['bug.js'],
    plugins: { jsdoc },
    rules: {
        'jsdoc/valid-types': 'warn'
    }
};
```

## ESLint sample

`bug.js`:

```js
/**
 * @typedef Test
 * @property {string} extends A value.
 */
```

## Environment

- Node version: v22.18.0
- ESLint version: 9.33.0
- `eslint-plugin-jsdoc` version: 54.1.0

**Comment by reduckted** [comment]
Thank you so much for the quick fix! ❤️
